tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -85,67 +82,112 @@
     private PlayerTokenResolver tokenResolver = new PlayerTokenResolver();
     private PlayerInitOptions initOptions;
 
-    public static KalturaPlayer createOVPPlayer(Context context, PlayerInitOptions initOptions) {
-        kalturaPlayerType = KalturaPlayerType.ovp;
-        KalturaPlayer kalturaPlayer = new KalturaPlayer(context, initOptions);
-        kalturaPlayer.serverUrl = KalturaPlayer.safeServerUrl(initOptions.serverUrl, KalturaPlayer.DEFAULT_OVP_SERVER_URL);
-        return kalturaPlayer;
+    public static void initializeOVP(Context context, int partnerId, @Nullable String serverUrl) {
+
+        PlayerConfigManager.retrieve(context, TVPlayerType.ovp, partnerId, serverUrl, (config, error, freshness) -> {
+           if (error != null) {
+               log.e("initialize KalturaPlayerType failed");
+           } else {
+               playerConfigRetreived = true;
+           }
+        });
     }
 
-    public static KalturaPlayer createOTTPlayer(Context context, PlayerInitOptions initOptions) {
-        kalturaPlayerType = KalturaPlayerType.ott;
-        KalturaPlayer kalturaPlayer = new KalturaPlayer(context, initOptions);
-        kalturaPlayer.serverUrl = KalturaPlayer.safeServerUrl(initOptions.serverUrl, null);
-        return kalturaPlayer;
+    public static void initializeOTT(Context context, int partnerId, @NonNull String serverUrl) {
+
+        PlayerConfigManager.retrieve(context, TVPlayerType.ott, partnerId, serverUrl, (config, error, freshness) -> {
+            if (error != null) {
+                log.e("initialize KalturaPlayerType failed");
+            } else {
+                playerConfigRetreived = true;
+            }
+        });
+    }
+
+    public static KalturaPlayer createOVPPlayer(Context context, PlayerInitOptions initOptions) throws KalturaPlayerNotInitializedException{
+        if ((playerConfigRetreived != null && playerConfigRetreived) || (initOptions != null && initOptions.tvPlayerParams != null)) {
```

Comment:
I think playerConfigRetreived should be a regular boolean. Unless there's a special meaning to the null value.